# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../uclibc/patches/psmisc-rpmatch.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Submitted By: Robert Connolly <robert at linuxfromscratch dot org> (ashes)
Date: 2005-03-20
Initial Package Version: 21.6
Upstream Status: Not submitted - Reported
ORIGIN: Gentoo psmisc-21.6-nonls.patch
Description: uClibc does not have rpmatch(). This patch creates a compatable function.

diff -Naur psmisc-21.6.orig/src/i18n.h psmisc-21.6/src/i18n.h
--- psmisc-21.6.orig/src/i18n.h	2004-11-09 11:25:15.000000000 +0000
+++ psmisc-21.6/src/i18n.h	2005-03-20 09:03:44.000000000 +0000
@@ -15,5 +15,13 @@
 #define _(String) (String)
 #endif
 
+#ifdef __UCLIBC__
+#define rpmatch(line) \
+	( (line == NULL) ? -1 : \
+	  (*line == 'y' || *line == 'Y') ? 1 : \
+	  (*line == 'n' || *line == 'N') ? 0 : \
+	  -1 )
+#endif
+
 #endif
 

